articles/build-an-outlook-clone-using-react-hooks/index.md

```diff
@@ -0,0 +1,881 @@
+
+```terminal
+npx create-react-app outlookclonereacthooks
+cd outlookclonereacthooks
+npm start
+```
+
+The folder structure should look like this.
+
+```
+outlookclonereacthooks
+├── README.md
+├── node_modules
+├── package.json
+├── .gitignore
+├── public
+│   ├── favicon.ico
+│   ├── index.html
+│   ├── logo192.png
+│   ├── logo512.png
+│   ├── manifest.json
+│   └── robots.txt
+└── src
+    ├── App.css
+    ├── App.js
+    ├── App.test.js
+    ├── index.css
+    ├── index.js
+    ├── logo.svg
+    └── serviceWorker.js
+```
+We will be working with class-based components in this article. The next article will talk about using Hooks for the same. A study of the two articles is enough to get a working understanding of React. Many examples that we find on GitHub are in class-based components. Having the ability to use React Hooks in place of class-based components is a useful skill to have.
+
+
+Let us create a folder called components in the root folder. The updated directory looks like this. Add the components *Email.js*, *EmailBody.js*,*SearchBox.js*,*Scroll.js*, and create directories named *styledComponents* and *css*. Add the files *ButtonStyles.js*, *EmailBody.js* and *EmailStyle.js* to styledComponents directory. Add file *EmailBody.css* to the css folder.  
+
+```
+outlookclonereacthooks
+├── README.md
+├── node_modules
+├── package.json
+├── .gitignore
+├── public
+│   ├── favicon.ico
+│   ├── index.html
+│   ├── logo192.png
+│   ├── logo512.png
+│   ├── manifest.json
+│   └── robots.txt
+└── src
+    ├── components     
+    │   ├── Email.js
+    │   ├── EmailBody.js
+    │   ├── Scroll.js
+    |   ├── styledComponents
+    |      ├── ButtonStyles.js
+    │      ├── EmailBodyStyles.js
+    │      ├── EmailStyle.js
+    |   ├── css
+    |      ├── EmailBody.css
+    ├── App.css
+    ├── App.js
+    ├── App.test.js
+    ├── index.css
+    ├── index.js
+    ├── logo.svg
+    └── serviceWorker.js
+```
+
+### Setting up API on localhost
+For the project, we need to make API calls to get the list of emails and email body. Thus, lets set that up. We will be using *json-server* package. This
+
+```
+npm install -g json-server
+```
+
+The single-line command installs the required package. -g denotes that the following package is installed globally.
+
+Our first JSON file consists of 15 emails. We will use My JSON Server, a hosted service that provides fake online REST APIs for free.
+
+The link for the hosted service is https://my-json-server.typicode.com/lalith1403/jsonemaillist/list
+
+The other API of interest is fetching the email body given the email id. Online hosting services have various bottlenecks when it comes to content length. Thus, let's use local hosting to get the data.
+
+The email body JSON file is as follows. Paste the following code into a file called *db.json*. Again *db.json* is a naming convention used.
```

Comment:
> Again *db.json* is a naming convention used.

Did you talk about this already? Either remove "Again" or add in the missing content that this is referring to.

Reply:
In the earlier article I had used db.json. I will refer to the previous article.